lib/Doctrine/DBAL/Driver/IBMDB2/Exception/PrepareFailed.php

```diff
@@ -13,8 +13,17 @@
  */
 final class PrepareFailed extends AbstractException
 {
-    public static function new(string $message): self
+    /**
+     * @psalm-param array{message: string}|null $error
+     */
+    public static function new(?array $error): self
     {
+        if ($error !== null) {
+            $message = $error['message'];
+        } else {
+            $message = 'Unknown error';
+        }
```

Comment:
Let's avoid the negative logic and the `else`:
```suggestion
        if ($error === null) {
            return new self('Unknown error');
        }
           
        return new self($error['message']);
```